spawnpool.py

```diff
@@ -205,25 +204,29 @@ def summarize(message, list):
             self._heart_beating = False
 
     @gen.coroutine
-    def _launch_container(self, path=None, enpool=True):
+    def _launch_container(self, user=None, enpool=True):
         '''Launch a new notebook server in a fresh container, register it with the proxy, and
         add it to the pool.'''
 
-        if path is None:
-            path = user_prefix()
+        if user is None:
+            user = new_user()
 
-        # This must match self.name_pattern or Bad Things will happen.
+        path = "user/" + user
```

Comment:
We should just only allow container names here, though if it borks for them it doesn't mess up everyone.

It is probably worth another PR.